src/Materials/PBR/pbrClearCoatConfiguration.ts

```diff
@@ -186,6 +215,8 @@ export class PBRClearCoatConfiguration {
     public prepareDefines(defines: IMaterialClearCoatDefines, scene: Scene): void {
         if (this._isEnabled) {
             defines.CLEARCOAT = true;
+            defines.CLEARCOAT_USE_ROUGHNESS_FROM_MAINTEXTURE = this._useRoughnessFromMainTexture;
+            defines.CLEARCOAT_TEXTURE_ROUGHNESS_IDENTICAL = this._texture !== null && this._texture._texture === this._textureRoughness?._texture;

```

Comment:
Is this good enough? What if the texture transform or sampling mode is different?

Reply:
The sampling mode is a property of `InternalTexture`, so that one is ok.

I had simplified the implementation to use the matrix transformation of the main texture if there is a main texture and the matrix transformation of the roughness texture if there is no main texture but a roughness texture.

Going to perform the changes to take into account the matrices from both textures.

Reply:
Sounds good